app/lib/registries/world_locations_registry.rb

```diff
@@ -0,0 +1,37 @@
+module Registries
+  class WorldLocationsRegistry
+    CACHE_KEY = "registries/world_locations".freeze
+
+    def all
+      @all ||= cached_locations
+    end
+
+    def [](slug)
+      all.find { |o| o['slug'] == slug }
+    end
+
+  private
+
+    def cached_locations
+      Rails.cache.fetch(CACHE_KEY, expires_in: 1.hour) do
+        fetch_locations
+      end
+    end
+
+    def fetch_locations
+      fetch_locations_from_worldwide_api.map { |result|
+        {
+          'title' => result.dig('title'),
+          'slug' => result.dig('details', 'slug')
+        }
+      }
+    end
+
+    def fetch_locations_from_worldwide_api
+      Services.worldwide_api.world_locations.with_subsequent_pages.to_a
+    rescue GdsApi::HTTPServerError, GdsApi::HTTPBadGateway
+      puts "ERROR: Could not get WorldLocation from Whitehall"
```

Comment:
Might be worth tweaking this to be "the API" or something rather than Whitehall?

I'm wondering what we want the behaviour to be here.  If it errors then we cache an empty array for an hour which is probably not quite what we want.  Might it be worth moving the `rescue` to outside the cache?

Reply:
Adding a test for this unhappy path would be amazing 👍

Reply:
Both very good points, thank you!